### PR TITLE
[RISCV] Limit max unroll factor to 8 by default

### PR DESCRIPTION
**Pull request by dcaballe**
We have been trying to selectively limit unrolling on RISC-V for those
cases that lead to register spilling. Unfortunately, it's becoming a
never-ending problem so this PR is limiting unrolling to 8 by default on
RISC-V while keeping unrolling of transpose operations disabled. That
seems to get rid of register spilling altogether. This sets a good
starting point for further unrolling tuning.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp [hunk 1]
```diff
@@ -198,36 +198,34 @@ static SmallVector<int64_t> getMinTilingSizesForEachDim(
...
     minTileSizes[fastestVaryingDim] =
         std::max<int64_t>(minTileSizes[fastestVaryingDim], tileSize);
   }
 
-  // Limit unrolling on transpose operations. For know, we assume the rightmost
-  // non-one tiled dimension is for vectorization and any other non-one
-  // dimension is for unrolling.
-  // TODO(dcaballe): Consider input and output transposes.
-  if (linalgOpInfo.isTranspose()) {
+  // Limit unroll factor. For know, we assume the rightmost non-one tiled
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp:206]: nit: typo `s/know/now`

**Review by MaheshRavishankar — APPROVED** [review]

**Review by hanhanW — COMMENTED** [review]
can we add a test to [materialize_riscv_launch_configuration.mlir](https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/LLVMCPU/test/materialize_riscv_launch_configuration.mlir)?

**Comment by hanhanW** [comment]
> can we add a test to [materialize_riscv_launch_configuration.mlir](https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/LLVMCPU/test/materialize_riscv_launch_configuration.mlir)?

nm, looks like it's an NFC?

**Review by hanhanW — APPROVED** [review]

**Comment by dcaballe** [comment]
> > can we add a test to [materialize_riscv_launch_configuration.mlir](https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/LLVMCPU/test/materialize_riscv_launch_configuration.mlir?rgh-link-date=2022-09-16T17%3A51%3A49Z)?
> 
> nm, looks like it's an NFC?

I have a WIP patch that is dumping the LinalgOpInfo and I was hoping to use that for testing. We should definitely add more tests about tile size computation but it's a bit difficult right now since the tile sizes are changed all over the place. I think more throughout testing should be addressed as part of [#10363](https://github.com/iree-org/iree/issues/10363)

**Pull request merged**